src/main/java/org/opentripplanner/routing/core/vehicle_sharing/VehiclePricingPackage.java

```diff
@@ -0,0 +1,243 @@
+
+    private BigDecimal parkingPricePerTimeTickInPackage;
+
+    private BigDecimal drivingPricePerTimeTickInPackageExceeded;
+
+    private BigDecimal parkingPricePerTimeTickInPackageExceeded;
+
+    private BigDecimal kilometerPrice;
+
+    private int secondsPerTimeTickInPackage;
+
+    private int secondsPerTimeTickInPackageExceeded;
+
+    private BigDecimal maxRentingPrice;
+
+    private boolean kilometerPriceEnabledAboveMaxRentingPrice;
+
+    public VehiclePricingPackage(){
+        /* By default creating a "no predefined package" configuration
+         * (package time limit is set to 0, so we only use the package exceeded properties to compute the price)
+         */
+        this(BigDecimal.ZERO, 0, 0, BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO, 60, 60, BigDecimal.ZERO, false);
+    }
+
+    public VehiclePricingPackage(BigDecimal packagePrice, int packageTimeLimitInSeconds, int freeSeconds, BigDecimal minRentingPrice, BigDecimal startPrice, BigDecimal drivingPricePerTimeTickInPackage, BigDecimal parkingPricePerTimeTickInPackage, BigDecimal drivingPricePerTimeTickInPackageExceeded, BigDecimal parkingPricePerTimeTickPackageExceeded, BigDecimal kilometerPrice, int secondsPerTimeTickInPackage, int secondsPerTimeTickInPackageExceeded, BigDecimal maxRentingPrice, boolean kilometerPriceEnabledAboveMaxRentingPrice) {
+        this.packagePrice = packagePrice;
+        this.packageTimeLimitInSeconds = packageTimeLimitInSeconds;
+        this.freeSeconds = freeSeconds;
+        this.minRentingPrice = minRentingPrice;
+        this.startPrice = startPrice;
+        this.drivingPricePerTimeTickInPackage = drivingPricePerTimeTickInPackage;
+        this.parkingPricePerTimeTickInPackage = parkingPricePerTimeTickInPackage;
+        this.drivingPricePerTimeTickInPackageExceeded = drivingPricePerTimeTickInPackageExceeded;
+        this.parkingPricePerTimeTickInPackageExceeded = parkingPricePerTimeTickPackageExceeded;
+        this.kilometerPrice = kilometerPrice;
+        this.secondsPerTimeTickInPackage = secondsPerTimeTickInPackage>0?secondsPerTimeTickInPackage:1;
+        this.secondsPerTimeTickInPackageExceeded = secondsPerTimeTickInPackageExceeded>0?secondsPerTimeTickInPackageExceeded:1;
+        this.maxRentingPrice = maxRentingPrice;
+        this.kilometerPriceEnabledAboveMaxRentingPrice = kilometerPriceEnabledAboveMaxRentingPrice;
+    }
+
+    public BigDecimal computeStartPrice(){
+        return packagePrice.add(startPrice);
+    }
+
+    public BigDecimal computeTimeAssociatedPrice(BigDecimal currentStartPrice, BigDecimal currentTimePrice, BigDecimal currentDistancePrice, int totalDrivingTimeInSeconds){
+        BigDecimal previousTotalPrice = currentStartPrice.add(currentTimePrice).add(currentDistancePrice);
+        if (!isMaxRentingPriceUsed() || (isMaxRentingPriceUsed() && !isMaxRentingPriceExceeded(previousTotalPrice))) {
```

Comment:
to się da uprościć do `if (!isMaxRentingPriceUsed() || !isMaxRentingPriceExceeded(previousTotalPrice)) {`